app/code/Magento/Eav/Model/Entity/Attribute.php

```diff
@@ -253,6 +253,16 @@ public function beforeSave()
             );
         }
 
+        /**
+         * Check attribute_code for allowed characters
+         */
+        if (isset($this->_data['attribute_code'])
+            && !preg_match('/^[a-z]+[a-z0-9_]+$/', $this->_data['attribute_code'])) {
```

Comment:
that regex, also force us to have attribute_code with at least 2 chars. Maybe it is ok, I just want to make you aware of that.